File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for (idx, v) in enumerate(nums):
            diff = target - v
            if diff in seen:
                return [idx, seen[diff]]
            
            seen[v] = idx

File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    seen = {}
    for (const [i, v] of nums.entries()) {
        if (v in seen) {
            return [seen[v], i]
        }
        
        seen[target - v] = i
    }
};